### DISCUSSION TITLE
Is state_file_interval only supposed to be applied once or repeatedly ?

### DISCUSSION THREAD

**Discussion by douwe2** [MusicPlayerDaemon/MPD#1638] [Q&A]
Hi,
Is my understanding wrong that when you set state_file_interval to for example 10 seconds the state_file is updated every 10 seconds ?
On my raspberry pi (mpd version 0.22.4) it only does it once after a state change, but no ever again until the next state change.

**Comment by MaxKellermann** [comment]
It is applied repeatedly, but if nothing changes, MPD will not bother writing a new state file, because it would be the same as the previous one.

**Comment by douwe2** [comment]
thanks so much for the quick reply and the clarification.

I was hoping I could set it to one second, then when the raspberry pi is 
turned of it can resume playing where it left off when it is turned on 
again.

Now it pretty  much goes back to the beginning of the song (or audiobook 
chapter. and those a really long).

I would be grateful if you could   point me in the right direction where 
in the code I could implement this feature ?

thanks,

   Ewoud

**Reply by MaxKellermann** [reply]
Progress during playback of a single song is not considered a "state change" and thus will not trigger the save.
Saving once per second will add a lot of write load to your SD card, and will wear it faster than necessary.
The code is here: https://github.com/MusicPlayerDaemon/MPD/blob/master/src/StateFile.cxx

**Comment by geneticdrift** [comment]
> when the raspberry pi is off

The state file is also written when MPD exits normally.
Maybe the cause of losing the state is that the raspi is turned off instead of shutting down cleanly.
